addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxPlayerHandler.java

```diff
@@ -396,20 +396,28 @@ public void coverArtChangeEvent(String mac, String coverArtUrl) {
      * @return A RawType object containing the image, null if the content type could not be found or the content type is
      *         not an image.
      */
-    private RawType downloadImage(String url) {
-        if (StringUtils.isNotEmpty(url)) {
-            if (!IMAGE_CACHE.containsKey(url)) {
-                IMAGE_CACHE.put(url, () -> {
-                    logger.debug("Trying to download the content of URL {}", url);
-                    return HttpUtil.downloadImage(url);
-                });
-            }
-            RawType image = IMAGE_CACHE.get(url);
-            if (image == null) {
-                logger.debug("Failed to download the content of URL {}", url);
-                return null;
-            } else {
-                return image;
+    private RawType downloadImage(String mac, String url) {
+        // Only get the image if this is my PlayerHandler instance
+        if (isMe(mac)) {
+            if (StringUtils.isNotEmpty(url)) {
+                if (!IMAGE_CACHE.containsKey(url)) {
+                    IMAGE_CACHE.put(url, () -> {
+                        logger.debug("Trying to download the content of URL {}", url);
+                        try {
+                            return HttpUtil.downloadImage(url);
+                        } catch (IllegalArgumentException e) {
```

Comment:
Strange error to get on this call, are you sure?

Reply:
Yes. I got this during my testing. I'll go back and refresh my memory on the specific cause.

Reply:
When passed this as a URL, `plugins/Pandora/html/images/icon.png`, it generates the following exception. The way the code currently is structured, it's highly unlikely that it would ever generate this exception. Nonetheless, I'd prefer to keep it, since IF it were to occur, it causes the player handler to fail silently.
```
2018-01-25 19:16:45.738 [ERROR] [ebox.handler.SqueezeBoxPlayerHandler] - IllegalArgumentException when downloading image from plugins/Pandora/html/images/icon.png
java.lang.IllegalArgumentException: Invalid URI host: null (authority: null)
        at org.eclipse.jetty.client.HttpClient.checkHost(HttpClient.java:487) ~[?:?]
        at org.eclipse.jetty.client.HttpClient.newHttpRequest(HttpClient.java:472) ~[?:?]
        at org.eclipse.jetty.client.HttpClient.newRequest(HttpClient.java:430) ~[?:?]
        at org.eclipse.jetty.client.HttpClient.newRequest(HttpClient.java:419) ~[?:?]
        at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrlAndGetReponse(HttpUtil.java:210) ~[?:?]
        at org.eclipse.smarthome.io.net.http.HttpUtil.downloadData(HttpUtil.java:454) ~[?:?]
        at org.eclipse.smarthome.io.net.http.HttpUtil.downloadImage(HttpUtil.java:416) ~[?:?]
        at org.eclipse.smarthome.io.net.http.HttpUtil.downloadImage(HttpUtil.java:387) ~[?:?]
        at org.eclipse.smarthome.io.net.http.HttpUtil.downloadImage(HttpUtil.java:373) ~[?:?]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxPlayerHandler.lambda$0(SqueezeBoxPlayerHandler.java:407) ~[?:?]
        at org.eclipse.smarthome.core.cache.ExpiringCache.refreshValue(ExpiringCache.java:81) [109:org.eclipse.smarthome.core:0.10.0.201801231340]
        at org.eclipse.smarthome.core.cache.ExpiringCache.getValue(ExpiringCache.java:61) [109:org.eclipse.smarthome.core:0.10.0.201801231340]
        at org.eclipse.smarthome.core.cache.ExpiringCacheMap.get(ExpiringCacheMap.java:186) [109:org.eclipse.smarthome.core:0.10.0.201801231340]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxPlayerHandler.downloadImage(SqueezeBoxPlayerHandler.java:414) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxPlayerHandler.coverArtChangeEvent(SqueezeBoxPlayerHandler.java:389) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler$SqueezeServerListener$14.updateListener(SqueezeBoxServerHandler.java:753) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler.updatePlayer(SqueezeBoxServerHandler.java:886) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler.access$8(SqueezeBoxServerHandler.java:870) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler$SqueezeServerListener.handleStatusMessage(SqueezeBoxServerHandler.java:750) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler$SqueezeServerListener.handlePlayerUpdate(SqueezeBoxServerHandler.java:531) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
        at org.openhab.binding.squeezebox.handler.SqueezeBoxServerHandler$SqueezeServerListener.run(SqueezeBoxServerHandler.java:409) [233:org.openhab.binding.squeezebox:2.3.0.201801260015]
```

Reply:
Then make the error a debug

Reply:
Agreed.  The only way it can happen is if artwork_url is corrupted such that it starts with http but is not a valid URL -- very unlikely to happen.